src/lib/snack-bar/snack-bar-ref.ts

```diff
@@ -47,6 +50,12 @@ export class MdSnackBarRef<T> {
     if (!this._afterClosed.closed) {
       this.containerInstance.exit();
     }
+    clearTimeout(this._dismissTimeoutHandle);
+  }
+
+  /** Dismisses the snack bar after some duration */
+  dismissAfter(duration: number): void {
```

Comment:
This should be an private method. If it were a public API and someone called it in combination with setting a duration, the earlier timeout ID would be lost and the issue this PR aims to resolve would still occur.